src/vs/workbench/contrib/terminal/common/terminalEnvironment.ts

```diff
@@ -269,6 +269,26 @@ export async function createTerminalEnvironment(
 			}
 		}
 
+		// Workaround for https://github.com/microsoft/vscode/issues/204005
+		// We should restore the following environment variables when a user
+		// launches the application using the CLI so that integrated terminal
+		// can still inherit these variables.
+		// We are not bypassing the restrictions implied in https://github.com/electron/electron/pull/40770
+		// since this only affects integrated terminal and not the application itself.
+		if (isMacintosh) {
+			// Restore NODE_OPTIONS if it was set
+			if (env['VSCODE_NODE_OPTIONS']) {
+				env['NODE_OPTIONS'] = env['VSCODE_NODE_OPTIONS'];
+				delete env['VSCODE_NODE_OPTIONS'];
```

Comment:
`sanitizeProcessEnvironment` will remove `VSCODE_` variables, this is not necessary but no harm in being explicit.

Reply:
We do this so as to not leak our own `NODE_OPTIONS` into terminal/debug processes where it could cause problems.